CMakeLists.txt

```diff
@@ -7,12 +7,17 @@ option(MBGL_WITH_SANITIZER "Use [address|thread|undefined] here" OFF)
 option(MBGL_WITH_RTTI "Compile with runtime type information" OFF)
 option(MBGL_WITH_OPENGL "Build with OpenGL renderer" ON)
 option(MBGL_WITH_EGL "Build with EGL renderer" OFF)
+option(MBGL_WITH_OSMESA "Build with OSMesa (Software) renderer" OFF)
 option(MBGL_WITH_WERROR "Make all compilation warnings errors" ON)
 
 if (MBGL_WITH_QT AND NOT CMAKE_OSX_DEPLOYMENT_TARGET)
     set(CMAKE_OSX_DEPLOYMENT_TARGET 13.0)
 endif()
 
+if(WIN32)
```

Comment:
This looks a bit intrusive. Qt for example should not use this toolchain. Is it too late if you load it in the platform-specific cmake file?

Reply:
I tried to make every if(WIN32) contained in the windows.cmake file, but this toolchain file needs to be included before the project command, otherwise it will be completely ignored. The idea here is to inject vcpkg if we are going to build in Windows platform. I'm heavily using vcpkg to simplify the dependencies.

Reply:
Do you think it's better to have the custom toolchain file to be included in cmake command? Something like:

```
cmake . -B build -DCMAKE_TOOLCHAIN_FILE=platform/windows/custom-toolchain.cmake
```

Reply:
Sorry to unresolve this now, but this is not OK for Qt builds, even if it may work by accident.

Reply:
would adding "AND NOT MBGL_WITH_QT" resolve this or would you want something else?

Reply:
Toolchain files should generally not be included by default and can be just provided when you call CMake (this is for example usually done for Android and iOS). Can you rather adapt your Node.js building scripts in such a way?

Reply:
Is that like @tdcosta100  suggested a few posts above with specifying DCMAKE_TOOLCHAIN_FILE? Can you point to an example from the Android/iOS that we could see what you mean? I looked a bit through those worksflows but I didn't really see anything that looked like what you describe.

Reply:
really just looking at what can be done to finish this PR off. Maybe another switch that's off by default that we can turn on witht he cmake command.

Reply:
@ntadej , this has been changed in the current version to not apply to QT. So I'm still looking if this resolve this for you or what you would like instead. This has already been approved and tested, so please this is the one thing in the way of merge I would say.

Reply:
Indeed we can clean it up later.